conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -175,32 +180,52 @@ def _get_prefix_path(self):
             else self._dep.package_folder
         return root_folder.replace("\\", "/")
 
+    @staticmethod
+    def _is_var_match(var, content):
+        """
+        Matches any *.pc variable in the content given.
+        """
+        if content:
+            return re.search("^{}=".format(var), content, re.MULTILINE)
+
     def content(self, info):
         assert isinstance(info, _PCInfo) and info.cpp_info is not None
 
-        prefix_path = self._get_prefix_path()
-        version = info.cpp_info.get_property("component_version") or self._dep.ref.version
-        libdirs = _get_formatted_dirs(info.cpp_info.libdirs, prefix_path)
-        includedirs = _get_formatted_dirs(info.cpp_info.includedirs, prefix_path)
-        bindirs = _get_formatted_dirs(info.cpp_info.bindirs, prefix_path)
-        custom_content = info.cpp_info.get_property("pkg_config_custom_content")
-
+        def _get_custom_pc_variables(cpp_info):
+            """
+            Prune any value from the final *.pc file if any user is setting their own ones through
+            the ``pkg_config_custom_content`` variable.
+            """
+            custom_content = cpp_info.get_property("pkg_config_custom_content")
+            prefix_path = "" if self._is_var_match("prefix", custom_content) else \
+                self._get_prefix_path()
+            libdirs = [] if self._is_var_match("libdir", custom_content) else \
+                _get_formatted_dirs(cpp_info.libdirs, prefix_path)
+            includedirs = [] if self._is_var_match("includedir", custom_content) else \
+                _get_formatted_dirs(cpp_info.includedirs, prefix_path)
+            bindirs = [] if self._is_var_match("bindir", custom_content) else \
+                _get_formatted_dirs(cpp_info.bindirs, prefix_path)
```

Comment:
I think can have a cleaner implementation and cleaner output file
- Use a regex that capture all variable names in the file
- Filter the variable names out in the dict below (line 217)

Reply:
What I'm thinking is to let users put that property as a dict-like instead of a string. That will simplify everything.

Reply:
But that could be breaking, and free format is still necessary (that would be a new feature? but we still need to make it work with current text input, right?)

Reply:
Yes, absolutely! It's going to be only a second way to proceed.